test/gtest/uct/test_md.cc

```diff
@@ -751,68 +770,87 @@ UCS_TEST_SKIP_COND_P(test_md, dereg_bad_arg,
                      !check_reg_mem_type(UCS_MEMORY_TYPE_HOST) ||
                      !ENABLE_PARAMS_CHECK)
 {
-    static const size_t size       = 1 * UCS_MBYTE;
-    static const unsigned md_flags = UCT_MD_MEM_ACCESS_REMOTE_PUT |
-                                     UCT_MD_MEM_ACCESS_REMOTE_GET;
+    static const size_t size                        = 1 * UCS_MBYTE;
+    static const unsigned md_flags_remote_rma       =
+            UCT_MD_MEM_ACCESS_REMOTE_PUT | UCT_MD_MEM_ACCESS_REMOTE_GET;
+    static const std::vector<unsigned> md_flags_vec = {
+        md_flags_remote_rma,
+        UCT_MD_MEM_ACCESS_REMOTE_ATOMIC
+    };
+    auto is_unsupported_caps                        =
+            [this] (unsigned md_flags) {
+        return (md_flags & md_flags_remote_rma) ?
+               !check_caps(UCT_MD_FLAG_INVALIDATE_RMA) :
+               (md_flags & UCT_MD_MEM_ACCESS_REMOTE_ATOMIC) ?
+               !check_caps(UCT_MD_FLAG_INVALIDATE_AMO) : true;
+    };
```

Comment:
seems over complicated
maybe put the checks in helper function and call this helper function with different flags parameters?